File: 0540-single-element-in-a-sorted-array/0540-single-element-in-a-sorted-array.py
class Solution:
    def singleNonDuplicate(self, arr: List[int]) -> int:
        lo,hi=0,len(arr)-1
        while(lo<hi):
            mid=int((lo+hi)/2)
            if(mid%2==1):
                mid-=1
            if(arr[mid]==arr[mid+1]):
                lo=mid+2
            else:
                hi=mid
        return arr[lo]

File: 0540-single-element-in-a-sorted-array/0540-single-element-in-a-sorted-array.java
class Solution {
    public int singleNonDuplicate(int[] arr) {
        int lo=0;
        int hi=arr.length-1;
        while(lo<hi){
            int mid=(lo+hi)/2;
            if(mid%2==1) mid--;
            if(arr[mid]==arr[mid+1]) lo=mid+2;
            else hi=mid;
        }
        return arr[lo];
    }
}